scripts/worker.py

```diff
@@ -166,7 +166,8 @@ def train_worker(gpu, ngpu_per_node, args, timers=None):
 
     """
     # fix random seed so models have the same starting weights
-    torch.manual_seed(42)
+    if args.seed is not None and args.seed > 0:
+        torch.manual_seed(args.seed)
```

Comment:
Since torch.manual_seed is set in main, is this needed again?

Reply:
I'm not sure if we need to set it everytime torch is imported. :-/ I didn't find much documentation supporting either theories. I left it in there for safety. Happy to learn more.

Reply:
I'm not sure either. Would be good to find out. Can probably also try to fetch the manual seed to see if it stays fixed in different parts of the code. The fewer place this has to be called the safer it is. My hunch is it might not be for each import, but might be once for each process

Reply:
I'm not sure there's an API to do that, didn't find any in the pytorch documentation

Reply:
re-setting the seed in the workers is not affecting the determinism of the models anyway. The models are still non-deterministic (binary files differ). So, I can remove them for now and when we investigate about determinism, we can think more about it. sound good ?

Reply:
@tijyojwad removed the seed setting fro mworkers. Added cuda.manual_seed() also in scripts/main.py.

Reply:
Thanks @ntadimeti !